### ISSUE TITLE
qx.event.Timer: false as default for "enabled" property

### ISSUE THREAD

**Issue by sbhtw** [qooxdoo/qooxdoo#9407]
The `enabled` property of class `qx.event.Timer` is set to `true` in property definition and then set to `false` in constructor

Property definition:

```javascript
    enabled :
    {
      init : true,
      check : "Boolean",
      apply : "_applyEnabled"
    },
```

Constructor:

```javascript
  construct : function(interval)
  {
    this.base(arguments);

    this.setEnabled(false);

    // ...
```

Why not just remove `setEnabled(false)` from constructor and specify `init: false` in property definition?

This would also fix the auto-generated documentation for the `enabled` property, which states `true` as default.

**Comment by johnspackman** [comment]
That sounds reasonable, please can you submit a PR with the change?

**Comment by sbhtw** [comment]
Ok.

Where can I find more information how to contribute to qooxdoo? The link in the "Contributing" section of https://github.com/qooxdoo/qooxdoo/blob/master/readme.rst is broken...

**Comment by johnspackman** [comment]
You start by cloning the qooxdoo/qooxdoo repo into your own user account, and then check it out on your computer; create a new branch and make the changes, and commit them and then push the changes back to your sbhtw/qooxdoo repo.

When you log back into github, you'll be able to create a pull request that will automatically be sent to the group for review.  That's it!

Here's some info on creating pull requests: https://help.github.com/articles/creating-a-pull-request/

When a PR is created, Travis will automatically run lint checks and the unit tests on your branch.  To make sure these work fine on your code before you commit your changes, in the `framework` directory of the repo run: 
```
$ ./generate.py test-source
$ ./generate.py lint
```

**Comment by level420** [comment]
@sbhtw you have to take into account, that the `enabled` property has an applyer `_applyEnabeld` defined which is called if the property value changes. In this case we have a property init value `true` which is changed to `false` in the constructor.

This means that the following code in the applyer gets executed:
```javascript
        window.clearInterval(this.__intervalHandler);
        this.__intervalHandler = null;
```
I don't know if this code has any effect, as `this._intervalHandler` should be `null`.

Changing the default of the property to `false` should be tested carefully.

And in theory this is a backwards compatibility breaking change, which may affect code deriving from `qx.event.Timer`. But to be honest, I don't see a use case for deriving from `qx.event.Timer`.

**Comment by cajus** [comment]
@level420 I don't think that this will be a problem. Looks like the default value is just used to trigger `_applyEnabled` - as it happens in the constructor. If the default is what it becomes anyway, using `this.initEnabled()` would be more straight forward.

**Comment by level420** [comment]
@sbhtw here is a short description on how you can create a PR as fast and as simple as possible. All in the browser, no installation locally needed. Give it a try:
https://github.com/qooxdoo/qooxdoo/issues/9340#issuecomment-308094551

**Comment by johnspackman** [comment]
@sbhtw my last comment was incomplete - the `./generate.py test-source` only builds the test runner application, you then need to open `test/index-source.html` in the browser; you need to do this via a small web server - any web server will do, it's just that you cannot run the unit tests via a  `file://` URL

**Comment by sbhtw** [comment]
See PR #9412  :-) Thanks for your help!

Should I leave this issue open?

**Comment by level420** [comment]
Fixed via #9412
